### ISSUE TITLE
etcd auto compact

### ISSUE THREAD

**Issue by sh0rez** [kontena/pharos-cluster#966]
**What would you like to be added**:
Run `etcd` with `--auto-compaction-retention=1` for complete auto compaction

**Why is this needed**:
Reduces storage footprint and improves overall performance

**Comment by jakolehm** [comment]
Kubernetes apiserver should compact etcd automatically:

```

--etcd-compaction-interval duration     Default: 5m0s
The interval of compaction requests. If 0, the compaction request from apiserver is disabled.

```

**Comment by sh0rez** [comment]
Yes, I already read that. But why is my database just increasing over time without me adding data after all?

**Comment by Timer** [comment]
@sh0rez you probably need to run a defrag, see https://github.com/kontena/pharos-cluster/issues/888

**Comment by sh0rez** [comment]
@Timer no, defrag did not help. Only by manually compacting AND running a defrag afterwards, the DB size decreased by nearly 60%
